platform/src/main/java/org/stellar/anchor/platform/event/EventProcessorManager.java

```diff
@@ -0,0 +1,163 @@
+      if (eventProcessorConfig.getClientStatusCallback().isEnabled()) {
+        clientsConfig
+            .getClients()
+            .forEach(
+                (clientConfig) -> {
+                  String processorName = null;
+                  switch (clientConfig.getType()) {
+                    case CUSTODIAL:
+                      processorName =
+                          CLIENT_STATUS_CALLBACK_EVENT_PROCESSOR_NAME_PREFIX
+                              + clientConfig.getSigningKey();
+                      break;
+                    case NONCUSTODIAL:
+                      processorName =
+                          CLIENT_STATUS_CALLBACK_EVENT_PROCESSOR_NAME_PREFIX
+                              + clientConfig.getDomain();
+                      break;
+                    default:
+                      errorF("Unknown client type: {}", clientConfig.getType());
+                  }
+                  processors.add(
+                      new EventProcessor(
+                          processorName,
+                          EventQueue.TRANSACTION,
+                          new ClientStatusCallbackHandler(clientConfig)));
+                });
+      }
+    }
+
+    // Start all the processors
+    processors.forEach(EventProcessor::start);
+  }
+
+  @PreDestroy
+  public void stop() {
+    for (EventProcessor processor : processors) {
+      try {
+        processor.stop();
+      } catch (AnchorException ex) {
+        errorF("Failed to shutdown event processor: {}}", processor.getClass());
+        errorEx(ex);
+      }
+    }
+  }
+
+  class EventProcessor implements Runnable {
+    private final String name;
+    private final EventQueue eventQueue;
+    private final EventHandler eventHandler;
+
+    private final ScheduledExecutorService consumerScheduler =
+        DaemonExecutors.newScheduledThreadPool(1);
+    private ScheduledFuture<?> processingTask = null;
+    private boolean stopped = false;
+
+    public EventProcessor(String name, EventQueue eventQueue, EventHandler eventHandler) {
+      this.name = name;
+      this.eventQueue = eventQueue;
+      this.eventHandler = eventHandler;
+    }
+
+    public void start() {
+      processingTask = consumerScheduler.scheduleWithFixedDelay(this, 1, 2, TimeUnit.SECONDS);
+    }
+
+    public void stop() throws AnchorException {
+      stopped = true;
+
+      if (processingTask != null) {
+        processingTask.cancel(true);
+      }
+
+      consumerScheduler.shutdown();
+    }
+
+    @SneakyThrows
+    @Override
+    public void run() {
+      infoF(
+          "The EventProcessor Kafka listening task is starting for the {} time.",
+          getConsumerRestartCount() + 1);
+      Session queueSession = eventService.createSession(name, eventQueue);
+      try {
+        while (!Thread.currentThread().isInterrupted() && !stopped) {
+          ReadResponse readResponse = queueSession.read();
+          List<AnchorEvent> events = readResponse.getEvents();
```

Comment:
Should we introduce some rate limiting when `getEvents` is 0? Maybe an exponential backoff?

Reply:
The Kafka `read()` is blocking and has a `poll()` timeout. It is left to the implementing class.